Algorithm.Framework/Alphas/PairsTradingAlphaModel.cs

```diff
@@ -96,38 +67,164 @@ public override IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slic
         /// <param name="changes">The security additions and removals from the algorithm</param>
         public override void OnSecuritiesChanged(QCAlgorithmFramework algorithm, SecurityChanges changes)
         {
-            foreach (var added in changes.AddedSecurities)
+            NotifiedSecurityChanges.UpdateCollection(_securities, changes);
+
+            UpdatePairs(algorithm);
+
+            // Remove pairs that has assets that were removed from the universe 
+            foreach (var security in changes.RemovedSecurities)
             {
-                // this model is limitted to looking at a single pair of assets
-                if (added.Symbol != _asset1 && added.Symbol != _asset2)
-                {
-                    continue;
-                }
+                var assetId = security.Symbol.ID.ToString();
+                var keys = _pairs.Keys.Where(k => k.Contains(assetId));
```

Comment:
We shouldn't be using string comparisons with symbols. Can this be changed to something like:
```
var keys = _pairs.Values.Where(p => p.Asset1 == security.Symbol || p.Asset2 == security.Symbol);
```
perhaps refactored into a method, `Pair.Contains(Symbol)` for similar syntax as the existing code.